lib/state/appStore.ts

```diff
@@ -135,9 +134,14 @@ export const useAppStore = create<AppState>()((set, get) => {
 
     secureStorage.removeItem(getWalletKey(wallets.length - 1));
 
-    get().setSelectedWalletId(0);
+    if (walletId === selectedWalletId) {
+      get().setSelectedWalletId(0);
+    } else if (walletId < selectedWalletId) {
```

Comment:
is this correct?

Reply:
EDIT: I think it's ok

Reply:
You could just always set the selected wallet to 0 though, it's probably just as good and simpler?

Reply:
We want to decrease the selected wallet id by one if the wallet id being deleted is less than it right?

Number of wallets: 3
Selected Id: 2
Wallet Id being removed: 1

As removing moves all existing wallets after it by 1, selected wallet id will now be 1

Reply:
> You could just always set the selected wallet to 0 though, it's probably just as good and simpler?

Yeah I thought it's good to do this as it might cause some confusion